### ISSUE TITLE
Bug: [no-unused-vars] reexporting enum does not pass linting

### ISSUE THREAD

**Issue by sviat9440** [typescript-eslint/typescript-eslint#5944]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=HYQwtgpgzgDiDGEAEAVaAXJBvAsAKCSQgA8YB7AJ0wmAFcwkBldEdZXAwpAQQBp8uSAEL9OhAMKjCAX3yy8+EuSpJQkWAmRoo6AEzYBRUpUwBLMMszNWyALyoMAOmtsA3HPxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtcNhyJo0DtEiyIAXVkBfEFqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
namespace Test {
  export enum State {
    A,
    B,
    C,
  }
}

export namespace Test2 {
  export import State = Test.State;
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "error"
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors

### Actual Result

'State' is defined but never used.

### Additional Info

If you confirm the bug, I will fix it soon

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.42.1` |
| `@typescript-eslint/parser` | `5.42.1` |
| `TypeScript` | `4.8.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by Josh-Cena** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/4129

**Comment by Josh-Cena** [comment]
This should be fixed by changing the AST shape (in 6.0). See https://github.com/typescript-eslint/typescript-eslint/issues/4130
